src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -207,14 +208,11 @@ private[internal] trait TypeMaps {
     def apply(tp: Type): Type = { traverse(tp); tp }
   }
 
-  abstract class TypeTraverserWithResult[T] extends TypeTraverser {
-    def result: T
-    def clear(): Unit
-  }
-
   abstract class TypeCollector[T](initial: T) extends TypeTraverser {
     var result: T = _
-    def collect(tp: Type) = {
+    // TODO This probably could be done better - without synchronization
```

Comment:
If umad was giving you warning for local var inside traverser I'm not convinced it shouldn't be `ThreadLocal` instead.
And I can bet it is right that there is violation :P